File: app/src/main/java/com/aminography/redirectglide/sampleapp/MyApiCallGlideUrl.java
package com.aminography.redirectglide.sampleapp;

import com.aminography.redirectglide.BaseApiCallGlideUrl;
import com.bumptech.glide.load.model.Headers;

import java.net.URL;

public class MyApiCallGlideUrl extends BaseApiCallGlideUrl {

    public MyApiCallGlideUrl(URL url) {
        super(url);
    }

    public MyApiCallGlideUrl(String url) {
        super(url);
    }

    public MyApiCallGlideUrl(URL url, Headers headers) {
        super(url, headers);
    }

    public MyApiCallGlideUrl(String url, Headers headers) {
        super(url, headers);
    }

    @Override
    public String extractImageUrl(String apiResult) {
        return MyApiDataModel.getSourceUrl(apiResult);
    }

}

File: sample-app/src/main/java/com/aminography/redirectglide/sampleapp/MyApiCallGlideUrl.kt
package com.aminography.redirectglide.sampleapp

import com.aminography.redirectglide.BaseApiCallGlideUrl
import com.aminography.redirectglide.sampleapp.MyApiDataModel.Companion.getSourceUrl
import com.bumptech.glide.load.model.Headers
import java.net.URL

class MyApiCallGlideUrl : BaseApiCallGlideUrl {

    constructor(url: URL?) : super(url)

    constructor(url: String?) : super(url)

    constructor(url: URL?, headers: Headers?) : super(url, headers)

    constructor(url: String?, headers: Headers?) : super(url, headers)

    override fun extractImageUrl(apiResult: String?): String? {
        return getSourceUrl(apiResult)
    }
}